content/nginx-one/connect-instances/connect-using-squid.md

```diff
@@ -0,0 +1,178 @@
+- [NGINX Agent is installed]({{< ref "nginx-one/agent/install-upgrade/" >}})
+- Access to the NGINX One console
+
+## Install Squid
+
+Follow the instructions in the [Squid website](https://wiki.squid-cache.org/SquidFaq/BinaryPackages) to install Squid on your server.
+
+
+## Configure Squid
+
+Follow the steps below to configure Squid with basic authentication.
+
+1. Open the Squid configuration file with your favorite text editor (you might need superuser privileges):
+
+   ```sh
+   vi /etc/conf/squid.conf
+   ```
+
+1. Add the following lines to configure the proxy settings:
+
+   ```conf
+   # Standard HTTP port for the proxy.
+   http_port myproxy.example.com:3128
+
+   # Define an ACL for allowing access from the agent's IP address
+   acl agent_ip src <AGENT_IP_ADDRESS>
+
+   # Allow the agent to connect to NGINX One Console
+   acl mgmt_server dstdomain agent.connect.nginx.com
+
+   # Allow HTTPS traffic (port 443 is default for HTTPS)
+   acl ssl_ports port 443
+
+   # HTTP access rules (allow the agent to access the destination server through the proxy)
+   http_access allow agent_ip mgmt_server ssl_ports
+
+   # Deny all other traffic by default (best practice)
+   http_access deny all
+   ```
+
+
+1. Save the changes and exit the text editor.
+1. Restart the Squid service to apply the changes:
+
+   ```sh
+   sudo systemctl reload squid
+   ```
+
+---
+
+## NGINX Agent Proxy configuration
+
+1. Open a secure connection to your instance using SSH and log in.
+1. Open the NGINX Agent configuration file (/etc/nginx-agent/nginx-agent.conf) with a text editor. To edit this file you need superuser privileges.
+1. Add or modify the `proxy` section to include the Squid proxy URL and timeout settings:
+
+   ```conf
+   server:
+      host: agent.connect.nginx.com
+      port: 443
+      proxy:
+         url: "http://proxy.example.com:3128"
+   ```
+
+1. Reload NGINX Agent to apply the changes:
+
+    ```sh
+    sudo systemctl restart nginx-agent
+    ```
+
+### In a containerized environment
+
+To configure NGINX Agent in a containerized environment:
+
+1. Run the NGINX Agent container with the environment variables set as follows:
+
+   ```sh
+   sudo docker run \
+      --add-host "myproxy.example.com:host-gateway" \
+      --env=NGINX_AGENT_COMMAND_SERVER_PORT=443 \
+      --env=NGINX_AGENT_COMMAND_SERVER_HOST=agent.connect.nginx.com \
+      --env=NGINX_AGENT_COMMAND_AUTH_TOKEN="<your-data-plane-key-here>" \
+      --env=NGINX_AGENT_COMMAND_TLS_SKIP_VERIFY=false \
+      --env=NGINX_AGENT_COMMAND_SERVER_PROXY_URL=http://myproxy.example.com:3128 \
+      --restart=always \
+      --runtime=runc \
+      -d private-registry.nginx.com/nginx-plus/agentv3:latest
+   ```
+
+
+## NGINX Agent proxy authentication
+
+If your Squid proxy requires authentication, you can specify the username and password in the `proxy` section of the `agent.conf` file:
+
+1. Open a secure connection to your instance using SSH and log in.
```

Comment:
It's very basic, but could have an SSH command here with a placeholder for the IP.